File: Tellma/ClientApp/src/app/data/entities/resource-definition-report-definition.ts
// tslint:disable:variable-name
// tslint:disable:max-line-length
import { EntityForSave } from './base/entity-for-save';

export interface ResourceDefinitionReportDefinitionForSave extends EntityForSave {
    ReportDefinitionId?: number;
    Name?: string;
    Name2?: string;
    Name3?: string;
}

export interface ResourceDefinitionReportDefinition extends ResourceDefinitionReportDefinitionForSave {
    ResourceDefinitionId?: number;
    SavedById?: number | string;
}

File: Tellma.Model.Application/ResourceDefinitionReportDefinition.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Tellma.Model.Common;

namespace Tellma.Model.Application
{
    [Display(Name = "ResourceDefinitionReportDefinition", GroupName = "ResourceDefinitionReportDefinitions")]
    public class ResourceDefinitionReportDefinitionForSave : EntityWithKey<int>
    {
        [Display(Name = "Definition_ReportDefinition")]
        [Required]
        public int? ReportDefinitionId { get; set; }
        public string Name { get; set; }
        public string Name2 { get; set; }
        public string Name3 { get; set; }
    }

    public class ResourceDefinitionReportDefinition : ResourceDefinitionReportDefinitionForSave
    {
        [Required]
        public int? Index { get; set; }

        [Required]
        public int? ResourceDefinitionId { get; set; }

        [Display(Name = "ModifiedBy")]
        [Required]
        public int? SavedById { get; set; }

        [Display(Name = "Definition_ReportDefinition")]
        [ForeignKey(nameof(ReportDefinitionId))]
        public ReportDefinition ReportDefinition { get; set; }

        // For Query

        [Display(Name = "ModifiedBy")]
        [ForeignKey(nameof(SavedById))]
        public User SavedBy { get; set; }
    }
}
